### PR TITLE
refactor: use ESQuery instead of raw queries

### PR DESCRIPTION
**Pull request by sravfeyn**
Stripped from the big ES7 PR

@millerdev @snopoke @calellowitz

### CODE REVIEW

**Review by dannyroberts — APPROVED** [review]
These are awesome. So much cleaner!

**Review by millerdev — COMMENTED** [review]
I don't know ES query syntax very well, so it's hard for me to evaluate if these are all equivalent. Do you feel like the test coverage is good enough to verify that?

File: corehq/apps/domain/calculations.py [hunk 1]
```diff
@@ -108,16 +108,10 @@ def cases_in_last(domain, days, case_type=None):
...
-            "modified_on": {
-                "from": then,
-                "to": now}}}}
-    query_params = {"domain.exact": domain, 'closed': False}
+    query = CaseES().domain(domain).modified_range(gte=then, lte=now).is_closed(False)
     if case_type:
-        query_params["type.exact"] = case_type
-    data = es_query(params=query_params, q=q, es_index='cases', size=1)
-    return data['hits']['total'] if data.get('hits') else 0
+        query.case_type(case_type)
```
Comment (esoergel) [inline corehq/apps/domain/calculations.py:113]:
```suggestion
        query = query.case_type(case_type)
```

File: corehq/apps/data_pipeline_audit/dbacessors.py [hunk 2]
```diff
@@ -255,6 +255,4 @@ def _get_user_base_doc_filter(doc_type):
         doc_type = doc_type[:-1]
 
     if doc_type == 'CommCareUser':
-        return {"term": {
-            "base_doc": "couchuser-deleted" if deleted else "couchuser"
-        }}
+        es.filters.term("base_doc", "couchuser-deleted" if deleted else "couchuser")
```
Comment (esoergel) [inline corehq/apps/data_pipeline_audit/dbacessors.py:258]:
```suggestion
        return es.filters.term("base_doc", "couchuser-deleted" if deleted else "couchuser")
```

File: custom/succeed/fields.py [hunk 3]
```diff
@@ -75,40 +74,17 @@ class PatientNameFilterMixin(object):
...
-            subterms.append({'or': filters})
-            es_filters["and"].append({'and': subterms} if subterms else {})
-
-        es_results = es_query(q=q, es_index='report_cases', dict_only=False)
-        return [(case['_source']['_id'], case['_source']['full_name']['#value']) for case in es_results['hits'].get('hits', [])]
+        if user.is_web_user():
+            owner_ids = [o.lower() for o in user.get_group_ids() if o]
+            if owner_ids:
+                query.owner(owner_ids)
+            query.user(user._id.lower())
```
Comment (esoergel) [inline custom/succeed/fields.py:84]: wow that old filter logic was bad.  `subterms` can't be falsy, but it's used in an if/else.  `and` is used to join a list, but that list contains only one filter.

However, I think the owner and user filters need to be joined with a `filters.OR`

**Review by esoergel — CHANGES_REQUESTED** [review]

File: corehq/apps/domain/calculations.py [hunk 4]
```diff
@@ -108,16 +108,10 @@ def cases_in_last(domain, days, case_type=None):
...
+    return query.run().total
```
[the remark this answers is not in the document]
Comment (sravfeyn) [reply corehq/apps/domain/calculations.py:114]: Thanks!

**Comment by sravfeyn** [comment]
Thanks for the great comments @esoergel ! Addressed all of them

**Comment by sravfeyn** [comment]
>  Do you feel like the test coverage is good enough to verify that?

@millerdev There is coverage for couple of them and the others don't have coverage. But these queries are pretty simple and straightforward, specially after Ethan's review

**Comment by millerdev** [comment]
@sravfeyn If some of them are not covered by tests then I think there is not adequate test coverage to verify that the results are the same for ES2 vs ES7. Would it be hard to add tests for queries that are currently untested?

**Review by esoergel — APPROVED** [review]

File: corehq/apps/es/forms.py [hunk 5]
```diff
@@ -71,7 +71,7 @@ def completed(gt=None, gte=None, lt=None, lte=None):
 
 
 def user_id(user_ids):
-    if not isinstance(user_ids, (list, set)):
+    if not isinstance(user_ids, (list, set, tuple)):
```
[the remark this answers is not in the document]
Comment (sravfeyn) [reply corehq/apps/es/forms.py:74]: I stumbled upon this in a ES7 test where the data was coming in tuples.

**Comment by sravfeyn** [comment]
> Would it be hard to add tests for queries that are currently untested?

@millerdev 
Except the custom report  query change (which hasn't been used in last 3 months and a bit hard to add test to) everything else is fairly obvious and small.  So, I am not very keen.

File: custom/succeed/fields.py [hunk 6]
```diff
@@ -75,40 +75,22 @@ class PatientNameFilterMixin(object):
...
+                query = query.filter(
+                    filters.OR(
+                        owner(owner_ids), user_filter(user._id.lower())
+                    )
+                )
+            else:
+                query = query.user(user._id.lower())
+
+        results = query.run().raw_hits
+        return [(case['_source']['_id'], case['_source']['full_name']['#value']) for case in results]
```
Comment (snopoke) [inline custom/succeed/fields.py:93]: you could avoid fetching the full case:

```suggestion
        results = query.values("full_name.#value")
        return [(case['_id'], case['full_name']['#value']) for case in results]
```

File: corehq/apps/domain/calculations.py [hunk 7]
```diff
@@ -126,17 +120,8 @@ def inactive_cases_in_last(domain, days):
...
-    q = {"query":
-             {"bool": {
-                 "must_not": {
-                     "range": {
-                         "modified_on": {
-                             "from": then,
-                             "to": now }}}}}}
-    data = es_query(params={"domain.exact": domain, 'closed': False}, q=q, es_index='cases', size=1)
-    return data['hits']['total'] if data.get('hits') else 0
+    return CaseES().domain(domain).modified_range(lt=then).is_closed(False).count()
```
[the remark this answers is not in the document]
Comment (esoergel) [reply corehq/apps/domain/calculations.py]: Actually I think `lt` is correct, though the docstring is wrong.  Looks like the docstring was copied from `cases_in_last` above.  This is `inactive_cases_in_last`, which is meant to be cases _not_ modified in however many days.  The raw query had a `must_not` wrapping the "from then to now" logic, equivalent to `.NOT(modified_range(gte=then))`

[the remark this answers is not in the document]
Comment (millerdev) [reply corehq/apps/domain/calculations.py]: @snopoke's suggestion was committed in https://github.com/dimagi/commcare-hq/pull/28264/commits/e64cb569e514b38f85e561eb6e72726673dd18c9. Should that be reverted @esoergel?

File: corehq/apps/data_pipeline_audit/dbacessors.py [hunk 8]
```diff
@@ -255,6 +255,4 @@ def _get_user_base_doc_filter(doc_type):
         doc_type = doc_type[:-1]
 
     if doc_type == 'CommCareUser':
-        return {"term": {
-            "base_doc": "couchuser-deleted" if deleted else "couchuser"
-        }}
+        return es.filters.term("base_doc", "couchuser-deleted" if deleted else "couchuser")
```
Comment (sravfeyn) [inline corehq/apps/data_pipeline_audit/dbacessors.py:258]: Haven't added a test, it's a very basic change.

File: corehq/apps/domain/calculations.py [hunk 9]
```diff
@@ -108,16 +108,10 @@ def cases_in_last(domain, days, case_type=None):
...
-                "from": then,
-                "to": now}}}}
-    query_params = {"domain.exact": domain, 'closed': False}
+    query = CaseES().domain(domain).modified_range(gte=then, lte=now).is_closed(False)
     if case_type:
-        query_params["type.exact"] = case_type
-    data = es_query(params=query_params, q=q, es_index='cases', size=1)
-    return data['hits']['total'] if data.get('hits') else 0
+        query = query.case_type(case_type)
+    return query.count()
```
Comment (sravfeyn) [inline corehq/apps/domain/calculations.py:114]: Added a test

File: corehq/apps/domain/calculations.py [hunk 10]
```diff
@@ -126,17 +120,8 @@ def inactive_cases_in_last(domain, days):
...
-    q = {"query":
-             {"bool": {
-                 "must_not": {
-                     "range": {
-                         "modified_on": {
-                             "from": then,
-                             "to": now }}}}}}
-    data = es_query(params={"domain.exact": domain, 'closed': False}, q=q, es_index='cases', size=1)
-    return data['hits']['total'] if data.get('hits') else 0
+    return CaseES().domain(domain).modified_range(gte=then).is_closed(False).count()
```
Comment (sravfeyn) [inline corehq/apps/domain/calculations.py:124]: Added a test

File: corehq/apps/domain/calculations.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (sravfeyn) [reply corehq/apps/domain/calculations.py]: I just came back to say this. I will revert. The docstring is wrong.

**Comment by sravfeyn** [comment]
@millerdev I have added tests where it makes sense. The custom report is bit more involved to test and hasn't been used since September 2019 and falls under SaaS and is also straightforward change for those with ES context. If you think this is not enough to add a test let me know.

File: corehq/apps/domain/calculations.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (snopoke) [reply corehq/apps/domain/calculations.py:124]: sorry for the confusion

**Review by millerdev — APPROVED** [review]

**Comment by millerdev** [comment]
> The custom report is bit more involved to test, hasn't been used since September 2019

Can we remove it?

**Comment by sravfeyn** [comment]
> Can we remove it?

Unfortunately we can't remove a custom report just based on usage. See Clayton's comment regarding this to a similar question of mine https://dimagi-dev.atlassian.net/browse/SAAS-10957

**Comment by sravfeyn** [comment]
Thanks for all the feedback guys.

**Pull request merged**